README.md

```diff
@@ -43,88 +43,22 @@ Write your distributed logic with the well-known fish and get the public state a
 ### Example
 
 ```js
-const MaterialRequest = ({id}: Props) => {
-  const [mrState] = useFish(MaterialRequestFish, id)
+const MaterialRequest = ({ id }: Props) => {
+  const mrState = useFish(MaterialRequestFish(id), id)
```

Comment:
```
useFish<T>(fish: Fish) => ReactFish;
useFish<T>((...:T) => Fish, props: T) => ReactFish;
useFish<T>(fish: Fish | (...:T) => Fish, props?: T) => ReactFish {
   // ...
  return [{
    props,
    state,
    feed,
  }, setProps(...props P)
}

```
as a nice side effect, you can use the setter to just pass in new props to the factory

I thought about, splitting it into two functions like useFish() and useStaticFish() could less magic.

I think useFish() and useFishFn() is not so cool, 'cause `useFishFn` should be the default in my opinion, but the `Fn` implied indirectly that this is the enhanced use-case

Reply:
Great idea!

Reply:
One thought in passing: who takes care to cancel the state subscription whenever we make a new one? If that is not explicitly canceled then we have a memory leak.

Reply:
Another thought: setProps seems like we’re adding yet one more layer of state handling that should already be handled by React. If the fish props are derived from a `useState` hook then React will take care to use the new fish for us already, right?

Reply:
We return a cleanup function from `React.useEffect` which cleans up the subscription.

I really like the ergonomics of @Alexander89 's proposal, as it closely resembles `React.useState`.

I pushed some changes, please give it a look.